hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -948,12 +1057,16 @@ int SaraNcpClient::waitReady(bool powerOn) {
         LOG(ERROR, "No response from NCP");
     }
 
-    if (!ready_) {
+    if (!ready_ && !firmwareUpdateR510_) {
         // Disable voltage translator
         modemSetUartState(false);
         // Hard reset the modem
         modemHardReset(true);
         ncpState(NcpState::OFF);
+        if (++waitReadyRetries >= 10) {
+            waitReadyRetries = 10;
```

Comment:
Let's say for example a modemEmergencyHardReset() doesn't actually make the modem `ready`, this could wrap in 1,670 some years and cause a problem.  I just add safety code like this to avoid unexpected issues, but I guess now that I've taken the time to make the calculation.. this will likely never occur... unless... it behaves differently than I just calculated... and then that one line of code that takes 8ns to run proves to be useful.  I can replace it with a comment if you'd like.

Reply:
I just did not see any place where it matters whether `waitReadyRetries` is 10 or 10000, hence my confusion.